bedrock/base/middleware.py

```diff
@@ -10,57 +10,174 @@
 
-    def __init__(self, get_response=None):
-        if not settings.USE_L10N:
-            warn(
-                """
-                The `USE_L10N` setting is False but LocaleURLMiddleware is
-                loaded. Consider removing bedrock.base.middleware.LocaleURLMiddleware
-                from your MIDDLEWARE setting.
-                """.strip()
+    def process_request(self, request):
+        if is_root_path_with_no_language_clues(request):
+            # Skip using Django's LocaleMiddleware
+            metrics.incr(
+                "bedrock.localemiddleware.skipdjangolocalemiddleware",
+                tags=[f"path:{request.path}"],
             )
-        self.get_response = get_response
+        else:
+            with normalized_get_language():
+                with simplified_check_for_language():
+                    return super().process_request(request)
 
-    def __call__(self, request):
-        response = self.process_request(request)
-        if response:
+    def process_response(self, request, response):
+        if is_root_path_with_no_language_clues(request):
+            # Skip using Django's LocaleMiddleware on the response cycle too
             return response
-        return self.get_response(request)
 
-    def process_request(self, request):
-        prefixer = urlresolvers.Prefixer(request)
-        urlresolvers.set_url_prefix(prefixer)
+        with normalized_get_language():
+            with simplified_check_for_language():
+                return super().process_response(request, response)
+
+
+@contextlib.contextmanager
+def normalized_get_language():
+    """
+    Ensures that any use of django.utils.translation.get_language()
+    within its context will return a normalized language code. This
+    context manager only works when the "get_language" function is
+    acquired from the "django.utils.translation" module at call time,
+    so for example, if it's called like "translation.get_language()".
+
+    Note that this does not cover every call to translation.get_language()
+    as there are calls to it on Django startup and more, but this gives us an
+    extra layer of (idempotent) fixing up within the request/response cycle.
+    """
+    get_language = translation.get_language
+
+    @wraps(get_language)
+    def get_normalized_language():
+        return normalize_language(get_language())
+
+    translation.get_language = get_normalized_language
+
+    try:
+        yield
+    finally:
+        translation.get_language = get_language
+
+
+@contextlib.contextmanager
+def simplified_check_for_language():
+    """Ensures that calls to trans_real.check_for_language within
+    its context will not check for the existence of files in
+    `appname/LANG_CODE/locale` and therefore avoid a false negative about
+    the langs we support (our lang files are in ./data/l10n-team/LANG_CODE)
+    but check_for_language is opinionated and expects only the Django directory
+    pattern for lang files."""
```

Comment:
**note:** This was a pain to find and patch. Does the docstring make enough sense?